lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/IntegrationExpansion.cxx

```diff
@@ -0,0 +1,195 @@
+    const OrthogonalBasis & basis,
+    const UnsignedInteger basisSize)
+  : FunctionalChaosAlgorithm(inputSample, weights, outputSample, distribution)
+  , basis_(basis)
+  , basisSize_(basisSize)
+{
+  // The arguments are checked in the base class, excepted the basis and its size
+  if (basis.getMeasure().getDimension() != distribution.getDimension()) throw InvalidArgumentException(HERE) << "Error: the basis must have a measure with the same dimension as the input distribution, heare measure dimension=" << basis.getMeasure().getDimension() << " and distribution dimension=" << distribution.getDimension();
+  if (basisSize == 0) throw InvalidArgumentException(HERE) << "Error: cannot project on a basis of size zero";
+}
+
+
+/* Virtual constructor */
+IntegrationExpansion * IntegrationExpansion::clone() const
+{
+  return new IntegrationExpansion(*this);
+}
+
+
+/* String converter */
+String IntegrationExpansion::__repr__() const
+{
+  return OSS() << "class=" << getClassName();
+}
+
+
+/* Computes the functional chaos */
+void IntegrationExpansion::run()
+{
+  // Check if we have to transform the input sample in order to adapt it
+  // to the basis
+  const Distribution measure(basis_.getMeasure());
+  const Bool noTransformation = (distribution_ == measure);
+  Sample transformedInputSample;
+  DistributionTransformation transformation;
+  if (noTransformation) transformedInputSample = inputSample_;
+  else
+    {
+      transformation = DistributionTransformation(distribution_, basis_.getMeasure());
+      transformedInputSample = transformation(inputSample_);
+    }
+  // Build the relevant functions from the basis
+  Indices indices(basisSize_);
+  indices.fill();
+  FunctionCollection functions(basisSize_);
+  for (UnsignedInteger i = 0; i < basisSize_; ++i)
+    functions[i] = basis_.build(i);
+  // Now build the design proxy and the least-squares method
+  DesignProxy designProxy(transformedInputSample, functions);
+  const Matrix designMatrix(designProxy.computeDesign(indices));
+  const Matrix transposedDesignMatrix(designMatrix.transpose());
+  const UnsignedInteger sampleSize = inputSample_.getSize();
+  const UnsignedInteger outputDimension = outputSample_.getDimension();
+  Matrix weightedOutput(sampleSize, outputDimension);
+  for (UnsignedInteger j = 0; j < outputDimension; ++j)
+    {
+      for (UnsignedInteger i = 0; i < sampleSize; ++i)
+        {
+          weightedOutput(i, j) = weights_[i] * outputSample_(i, j);
+        }
+    }
+  Point residuals(outputDimension);
+  Point relativeErrors(outputDimension);
+  Sample coefficients(basisSize_, outputDimension);
+  for (UnsignedInteger j = 0; j < outputDimension; ++j)
+    {
+      const Sample marginalOutputSample(outputSample_.getMarginal(j));
+      const Point rhs(marginalOutputSample.asPoint());
+      const Point coeffsJ(transposedDesignMatrix * rhs);
+      for (UnsignedInteger i = 0; i < basisSize_; ++i)
+        coefficients(i, j) = coeffsJ[i];
```

Comment:
I am missing something
What about the weights?
you generated above a `weightedOutput` but seems you are not using it!
`coefficients ?= transposedDesignMatrix * weightedOutput`

Reply:
:-) Good catch! I am currently rewriting the run() method in order to minimize the burden of converting matrices into samples. I will fix it on the way.

Reply:
Fixed